src/app/inventory/subclass.ts

```diff
@@ -26,6 +31,15 @@ interface V2SubclassInfo extends CommonSubclassInfo {
 }
 interface V3SubclassInfo extends CommonSubclassInfo {
   isV3: true;
+  plugCategoryHashes: {
+    supers: PlugCategoryHashes;
+    classAbilities: PlugCategoryHashes;
+    movementAbilities: PlugCategoryHashes;
+    melees: PlugCategoryHashes;
+    grenades: PlugCategoryHashes;
+    aspects: PlugCategoryHashes;
+    fragments: PlugCategoryHashes;
```

Comment:
Any reason this is manually curated to such an extent? It should be possible to derive these from the existing subclass hashes and socket category hashes.

Reply:
You're right, it's probably overkill. I couldn't figure this out at the time without pre-computing it using d2ai, but I'll take a closer look.

Reply:
`memoizeOne` is used in DIM for some very similar cases where d2ai isn't used.

https://github.com/DestinyItemManager/DIM/blob/7c1aaec83f889c145f7e1d89e4e173850d86303c/src/app/inventory/store/patterns.ts#L10-L23

Reply:
Thanks, this mapping is now dynamically built up in a memoized function:
https://github.com/DestinyItemManager/DIM/blob/540cff9ed790cc111aec8f9c49dfdd2bcea7bcd8/src/app/inventory/subclass.ts#L170-L211

Usages:

https://github.com/DestinyItemManager/DIM/blob/540cff9ed790cc111aec8f9c49dfdd2bcea7bcd8/src/app/inventory/subclass.ts#L330-L333

https://github.com/DestinyItemManager/DIM/blob/540cff9ed790cc111aec8f9c49dfdd2bcea7bcd8/src/app/utils/socket-utils.ts#L230-L234

Is that what you had in mind?

Reply:
Yes, exactly. Thanks! (I think DIM usually uses objects over `Map`s but I don't mind)